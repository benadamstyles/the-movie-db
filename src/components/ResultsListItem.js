import React from 'react'
import styled from 'react-emotion'
import PropTypes from 'prop-types'

const Button = styled('button')`
  background-color: transparent;
  background-image: url(${props =>
    `https://image.tmdb.org/t/p/original${props.imagePath}`});
  background-size: cover;
  background-position: center center;
  border: none;
  color: white;
  width: 100%;
  height: 6em;
  font-size: 4em;
  font-weight: bold;

  &:hover,
  &:focus {
    color: yellow;
  }
`

const ButtonText = styled('span')`
  text-shadow: 3px 3px 10px black;
`

export const Placeholder = Button.withComponent('p')

const ResultsListItem = props => (
  <li>
    <Button
      onClick={() => props.onSelectMovie(props.movie)}
      type="button"
      data-testid="result-button"
      imagePath={props.movie.backdrop_path || props.movie.poster_path}>
      <ButtonText data-testid="result-title">{props.movie.title}</ButtonText>
    </Button>
  </li>
)

ResultsListItem.propTypes = {
  movie: PropTypes.object.isRequired,
  onSelectMovie: PropTypes.func.isRequired,
}

export default ResultsListItem
