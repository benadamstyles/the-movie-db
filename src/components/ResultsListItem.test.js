import React from 'react'
import {render, cleanup, fireEvent} from 'react-testing-library'
import ResultsListItem from './ResultsListItem'

const mockMovie = {
  poster_path: '/cezWGskPY5x7GaglTTRN4Fugfb8.jpg',
  adult: false,
  overview: 'This is a mock movie.',
  release_date: '2012-04-25',
  genre_ids: [878, 28, 12],
  id: 0,
  original_title: 'Mock Movie',
  original_language: 'en',
  title: 'Mock Movie',
  backdrop_path: '/hbn46fQaRmlpBuUrEiFqv0GDL6Y.jpg',
  popularity: 7.353212,
  vote_count: 8503,
  video: false,
  vote_average: 7.33,
}

afterEach(cleanup)

describe('ResultsListItem', () => {
  it('handles clicks as expected', () => {
    const onSelectMovie = jest.fn()
    const {container, getByTestId} = render(
      <ResultsListItem movie={mockMovie} onSelectMovie={onSelectMovie} />
    )
    expect(container).toMatchSnapshot()
    fireEvent.click(getByTestId('result-title'))
    expect(onSelectMovie).toHaveBeenCalledWith(mockMovie)
  })
})
