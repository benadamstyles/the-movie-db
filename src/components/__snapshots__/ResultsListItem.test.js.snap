// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsListItem handles clicks as expected 1`] = `
<div>
  <li>
    <button
      class="css-rg16dp"
      data-testid="result-button"
      type="button"
    >
      <span
        class="css-17vo2iu"
        data-testid="result-title"
      >
        Mock Movie
      </span>
    </button>
  </li>
</div>
`;
